jj-fzf: 0.33.0 -> 0.34.0, fix missing runtime dependencies

razahussain81238 opened the pull request:
## Description

Fixes the jj-fzf package which was missing critical runtime dependencies and files. The current package only installs the main script, but jj-fzf requires several additional files to function properly.

## Changes

### Version Update
- Bumped version from 0.33.0 to 0.34.0

### Added Missing Dependencies
- **pandoc**: Required to generate man page during build
- **python3**: Required by `lib/gen-message.py` helper script
- **util-linux**: Provides `column` command used by the tool

### Build Process Improvements

#### postPatch Phase
- Fixed shebangs in main scripts (version.sh, preflight.sh, jj-fzf)
- Fixed shebangs in all lib/ directory scripts
- Patched Makefile.mk to use absolute path for env instead of /usr/bin/env

#### buildPhase
- Added proper build phase to generate documentation
- Runs `make doc/jj-fzf.1` to create manual page

#### installPhase Enhancements
- Installed preflight.sh helper script (sourced by main script)
- Installed version.sh helper script (referenced by --version flag)
- Installed complete lib/ directory with all helper scripts
- Installed generated manual page to proper location ($out/share/man/man1/)

#### Runtime Dependencies
- Updated wrapProgram to include python3 and util-linux in PATH

### Other Changes
- Changed platforms from `platforms.all` to `platforms.unix` (more appropriate for shell scripts)

## Testing

Successfully built and tested locally with `nix build .#jj-fzf`. Build log shows:
- All shebangs properly patched
- Manual page generated successfully
- All required files installed
- No runtime errors

## Impact

Without these fixes, jj-fzf fails at runtime with errors about missing files (lib/setup.sh, preflight.sh, etc.).

## CC

@bbigras (current maintainer)

Commit:
jj-fzf: 0.33.0 -> 0.34.0, fix missing runtime dependencies

The current jj-fzf package only installs the main script, but the
application requires several additional files to function properly:

- lib/ directory with helper scripts (sourced by main script)
- preflight.sh and version.sh (referenced by main script)
- Manual page documentation (needed for --man option)

This commit fixes the package by:

1. Version bump: 0.33.0 -> 0.34.0

2. Added missing build dependencies:
   - pandoc: required to generate man page from markdown
   - python3: required by lib/gen-message.py
   - util-linux: provides 'column' command used by the tool

3. Added postPatch phase to:
   - Fix shebangs in all shell scripts including lib/ directory
   - Patch Makefile.mk to use absolute path for env

4. Added buildPhase to generate documentation using make

5. Enhanced installPhase to install all required files:
   - preflight.sh and version.sh helper scripts
   - Complete lib/ directory with all helper scripts
   - Generated man page in proper location

6. Updated wrapProgram to include python3 and util-linux in PATH

7. Changed platforms from 'all' to 'unix' (more appropriate)

Without these fixes, jj-fzf fails at runtime with missing file errors.

razahussain81238 committed:
Update pkgs/by-name/jj/jj-fzf/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

SuperSandro2000 commented on pkgs/by-name/jj/jj-fzf/package.nix:
```diff
@@ -8,30 +8,95 @@
+
+    # Install documentation
+    mkdir -p $out/share/man/man1
+    install -D doc/jj-fzf.1 $out/share/man/man1/jj-fzf.1
```
```suggestion
    installManPage doc/jj-fzf.1 
```
from installShellFiles

razahussain81238 replied:
Done! Using `installManPage` helper from installShellFiles now.

SuperSandro2000 commented on pkgs/by-name/jj/jj-fzf/package.nix:
```diff
@@ -8,30 +8,95 @@
+
+    # Install lib directory and its contents
+    mkdir -p $out/bin/lib
+    cp -r lib/* $out/bin/lib/
```
$out/bin should not contain subdirectories. This should probably go to var or libexec or something similar

razahussain81238 committed:
fix(jj-fzf): apply review suggestions

This commit addresses the feedback from reviewers on the PR.

- Use `tag` instead of `rev` in `fetchFromGitHub`.
- Use `installManPage` helper to install the man page.
- Move build-time dependencies (`pandoc`, `python3`, `util-linux`) to `nativeBuildInputs`.
- Install helper scripts and the `lib` directory to `$out/libexec/jj-fzf` instead of `$out/bin`.
- Patch the main `jj-fzf` script to correctly locate its helper files in `libexec`.
- Updated `wrapProgram` to reflect the new `buildInputs`.

razahussain81238 committed:
Fix jj-fzf packaging: syntax fixes and update to 0.34.0

claude committed:
Fix nixpkgs formatting issues

- Format function arguments to multi-line for better readability
- Remove unnecessary semicolon in installPhase
- Add missing newline at end of file

These changes should resolve CI/CD formatting check failures.

razahussain81238 committed:
Merge pull request #2 from razahussain81238/claude/check-pr-status-011CUuj6F6W8TXT9HMX9tGhM

Fix nixpkgs formatting issues

razahussain81238 committed:
Add python3 and util-linux to buildInputs for runtime dependencies

razahussain81238 committed:
Fix syntax error: add missing colon after function parameters

razahussain81238 committed:
Fix lib.getExe syntax - use getExe' for specific binary

razahussain81238 commented on pkgs/by-name/jj/jj-fzf/package.nix:
```diff
@@ -8,30 +8,95 @@
     repo = "jj-fzf";
-    tag = "v${version}";
-    hash = "sha256-iVgX2Lu06t1pCQl5ZGgl3+lTv4HAPKbD/83STDtYhdU=";
+    rev = "v${version}";
```
Done! Changed to use `tag` attribute.

razahussain81238 committed:
Remove python3 and util-linux from nativeBuildInputs (only needed at runtime)

SuperSandro2000 commented on pkgs/by-name/jj/jj-fzf/package.nix:
```diff
@@ -8,49 +8,94 @@
+    # Patch scripts in lib directory
+    if [ -d lib ]; then
+      patchShebangs --build lib
+    fi
```
```suggestion
    # Patch scripts in lib directory
    patchShebangs --build lib
```
